pygmt/helpers/testing.py

```diff
@@ -0,0 +1,105 @@
+"""
+Helper functions for testing.
+"""
+
+import inspect
+import os
+
+from matplotlib.testing.compare import compare_images
+
+from ..exceptions import GMTImageComparisonFailure
+from ..figure import Figure
+
+
+def check_figures_equal(*, tol=0.0, result_dir="result_images"):
```

Comment:
One more thing about the result_dir is that, the `check_figures_equal` decorator generates images in `result_images` directory, while `pytest.mark.mpl_image_compare` generates images in directories like `results/tmpjtnnwqt4`.

Reply:
Yes, which was partly why I opened up the issue at https://github.com/matplotlib/pytest-mpl/issues/94, to get all of that `pytest-mpl` goodness (e.g. not having a hardcoded `result_dir`). I'll *try* to make a Pull Request to `pytest-mpl` for that, so we can just use a proper `@pytest.mark.mpl_check_equal` decorator in the future (will open a new issue after this one is merged). For now though, since we don't have many tests using `check_figures_equal` yet, we can probably just leave it like so.

Reply:
Yes, looks good to me.